bam/reader.go

```diff
@@ -101,6 +99,59 @@ const (
 	AllVariableLengthData        // Omit sequence, quality and auxiliary data.
 )
 
+// buffer is light-weight read buffer.
+type buffer struct {
+	off  int
+	data []byte
+}
+
+func (b *buffer) bytes(n int) []byte {
+	s := b.off
+	b.off += n
+	return b.data[s:b.off]
+}
+
+func (b *buffer) rest() []byte {
```

Comment:
I'd prefer it if we did something like

```
func (b *buffer) len() int {
    return len(b.data)-b.off
}
```

and then `rec.AuxFields = parseAux(b.bytes(b.len()))` where you are reading the tags.